loguru/_handler.py

```diff
@@ -175,7 +175,8 @@ def emit(self, record, level_id, from_decorator, is_raw, colored_message):
                     self._queue.put(str_record)
                 else:
                     self._sink.write(str_record)
-
+        except HandlerLockNotReentrant:
```

Comment:
I _think_ this shouldn't be re-raised.

If we swallow it here, we don't emit the log, but the object finalizer can complete whatever else it should do. If we raise it, it will likely still be swallowed (as is the case with finalizers), and the finalizer stop.

Reply:
So, I've been wondering about this, and I think we should not catch `HandlerLockNotReentrant` specifically. Instead, it can be handled by the `except Exception:` block bellow with conditional re-raising depending on `catch` argument.

It's true that it can cause `__del__` implementation to fail if the error is re-raised (because it won't be able to complete), but that's the expected behavior. Using the `logger` with `catch=False` means it can fail for different reasons, either because the lock is already in use or because the custom sink contains a bug for example.

Although the `__del__` documentation specifies raised exceptions are ignored by CPython, they do cause a warning to be displayed, so I don't think they will go unnoticed or be difficult to debug.

Finally, I think that this new safety measure will be triggered mostly when a user reuses the logger in a sink. In such case, the application should fail if the user specified `catch=False`.

Reply:
👍